### PR TITLE
Use 2 node cluster where indices are created with a replica.

### PR DESCRIPTION
**Pull request by naemono**
related #7387

`TestAPMKibanaAssociation` e2e test is also failing with `8.12.0-SNAPSHOT` with the following index/ds being created with a replica, so we must use a 2-node ES cluster.

```
yellow open .ds-metrics-fleet_server.agent_status-default-2023.12.18-000001 6GWxKTn8T2uYCVngquaUVg 1 1 3 0 26.6kb 26.6kb 26.6kb
```

Additionally, I've updated the Kibana `test/e2e/kb/version_upgrade_test.go` to fix some failures I was seeing simply running the tests.

TODO
- [x] Investigate additional test failures to see if any others need updating

image

### CODE REVIEW

File: test/e2e/kb/standalone_test.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func TestKibanaStandalone(t *testing.T) {
 	// set up a 1-node Kibana deployment manually connected to Elasticsearch
 	name := "test-kb-standalone"
 	esBuilder := elasticsearch.NewBuilder(name).
-		WithESMasterDataNodes(1, elasticsearch.DefaultResources).
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources). // TODO .ds-metrics-fleet_server.agent_status-default index is created with a replica
```
Comment (barkbay) [inline test/e2e/kb/standalone_test.go:71]: I'm not sure to understand the purpose of this TODO, and why `.ds-metrics-fleet_server.agent_status-default` is mentioned here (unless I'm missing something this tests only involves Kibana and Elasticsearch, not Agent?)

Comment (naemono) [reply test/e2e/kb/standalone_test.go:71]: Ya know, that's a very good question. From the diagnostics of the failure of this particular test:
cat_indices:
```
yellow open   .ds-metrics-fleet_server.agent_status-default-2023.12.18-000001 bS661iNXRBOHk8Qf6fs_2g   1   1         15            0    132.2kb        132.2kb false
```
Let me run locally and see what I can find.

Comment (naemono) [reply test/e2e/kb/standalone_test.go:71]: Verified locally that this index is being created with just kibana + elasticsearch:
```
yellow open .ds-metrics-fleet_server.agent_status-default-2023.12.19-000001 Lpe5MTJ8R6CNusmbk6gwrw 1 1 6 0 52.9kb 52.9kb 52.9kb
```

Will reach out to Kibana team.

Comment (rhr323) [reply test/e2e/kb/standalone_test.go:71]: I fixed a couple of these (seems like maybe not all?) Thibault opened an issue https://github.com/elastic/cloud-on-k8s/issues/7376 which has a reference to an issue I opened for fleet. I hope this is helpful.

Comment (naemono) [reply test/e2e/kb/standalone_test.go:71]: I verified with the Fleet/Kibana team, fleet integration is now enabled by default, so this index will exist with just vanilla Kibana+Elasticsearch. I'll update the comment/todo to be more clear. We still should be able to control num replicas sometime in the future and go back down to 0 replicas.

**Review by barkbay — APPROVED** [review]
LGTM, I'm just wondering if the `TODO` can be removed

File: test/e2e/apm/association_test.go [hunk 2]
```diff
@@ -61,7 +61,8 @@ func TestAPMKibanaAssociation(t *testing.T) {
 
 	esBuilder := elasticsearch.NewBuilder(name).
 		WithNamespace(ns).
-		WithESMasterDataNodes(1, elasticsearch.DefaultResources).
+		WithESMasterDataNodes(2, elasticsearch.DefaultResources). // TODO .ds-metrics-fleet_server.agent_status-default index is created with a replica.
+
```
[the remark this answers is not in the document]
Comment (rhr323) [reply test/e2e/apm/association_test.go:65]: I guess it was meant only as a temporary fix until https://github.com/elastic/integrations/issues/8671 would be fixed. 

Coincidentally, it seems the fix has just been merged: https://github.com/elastic/kibana/pull/173689, so hopefully this change can be reverted soon (although it's likely not a huge issue to run 2 node clusters).

[the remark this answers is not in the document]
Comment (naemono) [reply test/e2e/apm/association_test.go]: I've updated all TODO items with
`// TODO revert when https://github.com/elastic/cloud-on-k8s/issues/7376 is resolved.` which notes the Kibana PR that should resolve this.

**Pull request merged**